src/js/features/ingestion/ingestion.store.ts

```diff
@@ -0,0 +1,56 @@
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import axios from 'axios';
+import { lastIngestionsUrl } from '@/constants/configConstants';
+import { printAPIError } from '@/utils/error.util';
+import { ingestionData, LastIngestionResponse } from '@/types/lastIngestionResponse';
+
+export const makeGetIngestionDataRequest = createAsyncThunk<LastIngestionResponse, void, { rejectValue: string }>(
+  'ingestionData/getIngestionData',
+  (_, { rejectWithValue }) =>
+    axios
+      .get(lastIngestionsUrl)
+      .then((res) => res.data)
+      .catch(printAPIError(rejectWithValue))
+);
+
+export interface IngestionDataState {
+  isFetchingIngestionData: boolean;
+  ingestionData: ingestionData[];
+  lastEndTimesByDataType: { [dataType: string]: string };
+}
+
+const initialState: IngestionDataState = {
+  isFetchingIngestionData: false,
+  ingestionData: [],
+  lastEndTimesByDataType: {},
+};
+
+const IngestionDataStore = createSlice({
+  name: 'ingestionData',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(makeGetIngestionDataRequest.pending, (state) => {
+      state.isFetchingIngestionData = true;
+    });
+    builder.addCase(
+      makeGetIngestionDataRequest.fulfilled,
+      (state, { payload }: PayloadAction<LastIngestionResponse>) => {
+        state.ingestionData = payload;
+        payload.forEach((ingestion) => {
+          const dataType = ingestion.details.request.tags.workflow_metadata.data_type;
+          const endTime = ingestion.details.run_log.end_time;
```

Comment:
do you need to parse this from string to a date for comparison, or does string comparison work well enough?

Reply:
Agree with your point. While string comparison may work well given consistent date formatting, parsing should be safer.